src/app/core/routing/category/category-route.pipe.ts

```diff
@@ -1,12 +1,30 @@
-import { Pipe, PipeTransform } from '@angular/core';
+import { OnDestroy, Pipe, PipeTransform } from '@angular/core';
+import { Subject, takeUntil } from 'rxjs';
 
+import { ShoppingFacade } from 'ish-core/facades/shopping.facade';
 import { CategoryView } from 'ish-core/models/category-view/category-view.model';
+import { Category } from 'ish-core/models/category/category.model';
 
 import { generateCategoryUrl } from './category.route';
 
 @Pipe({ name: 'ishCategoryRoute', pure: true })
-export class CategoryRoutePipe implements PipeTransform {
+export class CategoryRoutePipe implements PipeTransform, OnDestroy {
+  private pathElements: Category[];
+
+  private destroy$ = new Subject<void>();
+
+  constructor(private shoppingFacade: ShoppingFacade) {}
   transform(category: CategoryView): string {
-    return generateCategoryUrl(category);
+    this.shoppingFacade
```

Comment:
I don't like this. First of all this will probably create many subscriptions that are never disposed of. 
Second, there's no guarantee that the right `pathElements` will be used in the return. 

Either:
- return an observable and chain with an `async` pipe. A bit verbose but clean
- there's probably some other solution using impure pipes that keeps track of the category state and marks itself for change detection when the categoryPathElements change. 

I would have to do some more research here but I'm sure there's a nice solution. 
Maybe something completely different exists too.

Reply:
Yeah! Your definetly right. :) A good solution for that is documented in the offical Angular docs. (https://angular.io/guide/pipes#caching-http-requests) It uses a caching feature to subscribe only then, when the input (category) changes. In addition the custom pipe mark itself for a change detection whenever the categoryPathElements$ observable emits a new value.

Reply:
Nice solution, but I have two more questions:
- Why do you mark for changes manually when you can use the `pure: false` attribute to do it for you (as shown in the docs)
- You still run into the issue of aggregating subscriptions. Since `categoryPathElements$` is a long running observable, unlike `httpClient.get`, you also need to unsubscribe from the old subscription whenever a new value is requested. Otherwise, you might have multiple long running subscriptions leaking.